src/core/dataflow/src/DataFlowEngine.cpp

```diff
@@ -406,35 +409,26 @@ DataFlowResult *DataFlowEngine::applyCustomizableForwardAnalysis(
         /*
          * Compute IN[i]
          */
-        auto &inSetOfI = df->IN(i);
+        auto &inSetOfI = getInSetOfInst(df, i);
```

Comment:
Why is this generalization needed?

Reply:
The use of `inSetOfI` is `computeIN(i, predI, inSetOfI, df);`, which is a lambda function updates the IN set based on predecessors.
The generalization is needed here is because the `computeIN` will be changed to `computeOUT`, `predI` will be changed to `succI` in backward. 
Therefore, the update of in set should be the out set instead